### DISCUSSION TITLE
Create database connection with pre-configured SSL socket

### DISCUSSION THREAD

**Discussion by jackwotherspoon** [PyMySQL/mysqlclient#642] [Q&A]
Is there the ability to create database connections from a pre-configured SSL Socket using `mysqlclient`?

Pymysql allows a sock argument to it's connect method ([code](https://github.com/PyMySQL/PyMySQL/blob/main/pymysql/connections.py#L626)) that allows passing in a pre-configured socket, wondering if mysqlclient has something equivalent?

I ask because the [Cloud SQL Python Connector](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector) would like to support `mysqlclient` as a driver option for connecting to Google Cloud SQL instances. Currently the only MySQL driver we support is pymysql ([code](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector/blob/main/google/cloud/sql/connector/pymysql.py#L53-L63)) as we need the ability to pass in a pre-configured SSL socket/connection.

Let me know, happy to give further details if need be or raise a feature request 😄

**Comment by methane** [comment]
mysqlclient is a binding of libmysqlclient.
Does any library using libmysqlclient supports that feature?

**Comment by jackwotherspoon** [comment]
@methane not that I am aware of, I will investigate a bit further into other C-binding libraries that leverage libmysqlclient. I guess that for `mysqlclient` to support such a feature, a patch to `libmysqlclient` would be required first?
